File: app/src/main/java/com/example/hellooboe/AdapterLoadFile.java
package com.example.hellooboe;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class AdapterLoadFile extends RecyclerView.Adapter<AdapterLoadFile.LoadFileItem> {

    private String[] fileNames;

    public AdapterLoadFile(String[] fileNames) {
        this.fileNames = fileNames;
    }

    @Override
    @NonNull
    public LoadFileItem onCreateViewHolder(ViewGroup parent, int viewType) {
        TextView textViewLoadFile = (TextView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.text_view_load_file, parent, false);
        return new LoadFileItem(textViewLoadFile);
    }

    @Override
    public void onBindViewHolder(LoadFileItem holder, int position) {
        holder.textViewFileName.setText(fileNames[position]);
    }

    @Override
    public int getItemCount() {
        return fileNames.length;
    }

    public static class LoadFileItem extends RecyclerView.ViewHolder {

        public TextView textViewFileName;

        public LoadFileItem(TextView textViewFileName) {
            super(textViewFileName);
            this.textViewFileName = textViewFileName;
        }

    }

}

File: app/src/main/java/io/fourth_finger/sound_sculptor/AdapterLoadFile.kt
package io.fourth_finger.sound_sculptor

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import io.fourth_finger.sound_sculptor.AdapterLoadFile.LoadFileItem

class AdapterLoadFile(private val fileNames: Array<String>) : RecyclerView.Adapter<LoadFileItem>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LoadFileItem {
        val textViewLoadFile = LayoutInflater.from(parent.context)
            .inflate(R.layout.text_view_load_file, parent, false) as TextView
        return LoadFileItem(textViewLoadFile)
    }

    override fun onBindViewHolder(holder: LoadFileItem, position: Int) {
        holder.textViewFileName.text = fileNames[position]
    }

    override fun getItemCount(): Int {
        return fileNames.size
    }

    class LoadFileItem(var textViewFileName: TextView) : RecyclerView.ViewHolder(
        textViewFileName
    )
}